python_modules/libraries/dagster-components/dagster_components/lib/dbt_project.py

```diff
@@ -43,28 +42,37 @@ class DbtProjectComponentTranslator(DagsterDbtTranslator):
     def __init__(
         self,
         *,
+        params: Optional[AssetAttributesModel],
         value_resolver: TemplatedValueResolver,
-        translator_params: Optional[DbtNodeTranslatorParams] = None,
     ):
+        self.params = params or AssetAttributesModel()
         self.value_resolver = value_resolver
-        self.translator_params = translator_params
+
+    def _get_rendered_attribute(
+        self, attribute: str, dbt_resource_props: Mapping[str, Any], default_method
```

Comment:
How about we type attribute as `Literal["key", "group_name", "tags", "automationc_condition"]` or what ever other ones I am missing?